biogeophys/FatesPlantHydraulicsMod.F90

```diff
@@ -840,6 +862,16 @@ subroutine UpdatePlantHydrLenVol(ccohort,site_hydr)
     roota    = prt_params%fnrt_prof_a(ft)
     rootb    = prt_params%fnrt_prof_b(ft)
 
+    dbh_max = EDPftvarcon_inst%allom_dbh_max(ft)
+    dbh_0   = EDPftvarcon_inst%allom_dbh_0(ft)
+    z_fr_max = EDPftvarcon_inst%allom_zfr_max(ft)
+
+    z_fr_0 = EDPftvarcon_inst%allom_zfr_0(ft)
+    frk    = EDPftvarcon_inst%allom_frk(ft)
+    dbh      =  ccohort%dbh
+    dbh_rev  = (dbh - dbh_0)/(dbh_max - dbh_0)
```

Comment:
Now I see here how dbh_max and dbh_0 are used, to calculate a sort of 'relative size', I wonder whether these should be estimated using the allometry functions that we already have. I worry that adding this many allometric parameters might mean that people tie themselves up in terrible knots when trying to parameterize the model. Further, we are only using dbh_max to calculate the rooting depth, when people might reasonably think that it also actually affects the size of the trees...  Do you think it's reasonable to calculate this metric from the allometry functions, @rgknox ?

Reply:
@rosiealice , I was thinking something similar.  We do have a routine that calculates "derived" parameters (somewhat of an oxymoron).  But they are essentially constants that are derived from user controlled constants.  So dbh_0 would be calculated from the initial cohort height.  Is it possible that dbh_max is similar to the dbh at maximum height parameter?

Reply:
I think dbh_max and dbh at maximum height are different concepts. dbh_max here is the maximum dbh a PFT can possible have in reality without any constrains, while dbh at maximum height, by my understanding, is the dbh the PFT will no longer increase height, but they can still increase dbh, right?

Reply:
Is there a paper that addresses the conversion from dbh to relative size for this purpose? Maybe that would help us elucidate the way in which it should be used. Note that in FATES, there is in fact no maximum DBH. Plants can just keep getting fatter, but generally not taller. So, whatever you set this variable as, it it will, I think, be possible to exceed it. Given the lack of data on rooting depths in general, my suspicion is that we do not know exactly how this scaling should work, and probably can't distinguish between the merits of different formats of relative size. (if not, then, great!)

Reply:
@rosiealice , good point. I was just thinking there should be a up limit when the size can no longer increases e.g. from basic biological processes and/or physics. So I borrow the idea of the logistic population growth model, so the dbh_max is analog to the carrying capacity. I should look into the literatures see if there is any studies on that. 
Yes, different formulas of relative size can generate exact the same shape, which is determined by the values of the parameters. 

I am hoping to have a root growth model that best represents the biological and physical mechanisms, in the way the formula represent the processes/interactions and each parameter has biological or physical meaning. Do you have any suggestions/ideas?

Reply:
I wanted to check in on this particular point. @JunyanDing.  As @rosiealice mentions, the growth equations in the core fates scaling algorithm don't really understand the concept of a maximum diameter.   I see two paths that are viable going forward.  1) We use "the diameter at maximum height" (parameter already exists)  instead of new parameter that defines the maximum diameter for relative root depth scaling.  Alternatively, 2) if we do want to have a separate parameter that defines the maximum carrying capacity,  I suppose it would be ok as long as we use a naming convention and use of metadata to explain its for the rooting depth and not used to actually constrain plant diameter. Part of my job is reduce the overall build-up of unused or duplicative parameters in the fates model (which is inevitable), so I'm always going to push for alternative 2.  :)

Also, @JunyanDing : How about dbh_0.  Can we get rid of this parameter and just use the diameter the cohorts start with when they are recruited (ie their smallest possible diameter)?

Reply:
@rgknox I like your idea of alternative 2 for max dbh,  :-) .maybe call it dbh_max_zroot?  As for dbh_0, I agree, we can use the smallest possible diameter here.

Reply:
the naming convention was resolved via #759